src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -2010,23 +2137,290 @@ private void resetVariable() {
+				radioWithMovement.setSelected(false);
+			}
+		});
+		return radioButtonAll;
+	}
+
+	private JRadioButton getMedicalWithNonZeroQuatityRadioButton() {
+		if (radioOnlyNonZero == null) {
+			radioOnlyNonZero = new JRadioButton(MessageBundle.getMessage("angal.inventory.medicalwithonlynonzeroqty.btn"));
+			radioOnlyNonZero.setMnemonic(MessageBundle.getMnemonic("angal.inventory.medicalwithonlynonzeroqty.btn.key"));
+		}
+		radioOnlyNonZero.addActionListener(actionEvent -> {
+			if (radioOnlyNonZero.isSelected()) {
+				radioButtonAll.setSelected(false);
+				radioWithMovement.setSelected(false);
+			}
+		});
+		return radioOnlyNonZero;
+	}
+
+	private JRadioButton getMedicalWithMovementRadioButton() {
+		if (radioWithMovement == null) {
+			radioWithMovement = new JRadioButton(MessageBundle.getMessage("angal.inventory.medicalwithmovementonly.btn"));
+			radioWithMovement.setMnemonic(MessageBundle.getMnemonic("angal.inventory.medicalwithmovementonly.btn.key"));
+		}
+		radioWithMovement.addActionListener(actionEvent -> {
+			if (radioWithMovement.isSelected()) {
+				radioButtonAll.setSelected(false);
+				radioOnlyNonZero.setSelected(false);
+			}
+		});
+		return radioWithMovement;
+	}
+
+	private JButton getCancelButton() {
+		if (jButtonCancel == null) {
+			jButtonCancel = new JButton(MessageBundle.getMessage("angal.common.cancel.btn"));
+			jButtonCancel.setMnemonic(MessageBundle.getMnemonic("angal.common.cancel.btn.key"));
+		}
+		jButtonCancel.addActionListener(actionEvent -> {
+			frame.dispose();
+		});
+		return jButtonCancel;
+	}
+	
+	private List<MedicalInventoryRow> loadNewInventoryTable(boolean withNonZeroQty, MedicalType medicalTypeSelected) throws OHServiceException {
+		List<MedicalInventoryRow> inventoryRowsList = getMedicalInventoryRows(null);
+		if (withNonZeroQty) {
+			inventoryRowsList = inventoryRowsList.stream().filter(inv -> inv.getTheoreticQty() > 0).collect(Collectors.toList());
+		}
+		if (medicalTypeSelected != null) {
+			inventoryRowsList = inventoryRowsList.stream()
+							.filter(inv -> inv.getMedical().getType().getDescription().equals(medicalTypeSelected.getDescription()))
+							.collect(Collectors.toList());
+		}
+		return inventoryRowsList;
+	}
+
+	private List<MedicalInventoryRow> loadNewInventoryTable(MedicalType medicalTypeSelected) throws OHServiceException {
+		List<MedicalInventoryRow> inventoryRowsList = getMedicalInventoryRowsWithMovement();
+		if (medicalTypeSelected != null) {
+			inventoryRowsList = inventoryRowsList.stream()
+							.filter(inv -> inv.getMedical().getType().getDescription().equals(medicalTypeSelected.getDescription()))
+							.collect(Collectors.toList());
+		}
+		return inventoryRowsList;
+	}
+
+	private JButton getOkButton() {
+		if (jButtonOk == null) {
+			jButtonOk = new JButton(MessageBundle.getMessage("angal.common.ok.btn"));
+			jButtonOk.setMnemonic(MessageBundle.getMnemonic("angal.common.ok.btn.key"));
+			jButtonOk.addActionListener(actionEvent -> {
```

Comment:
don't you realize that the three `if` blocks are identical now? The three options have same effect...

```
if (radioButtonAll.isSelected()) {
    // same logic with new InventoryRowModel(true);
}
if (radioOnlyNonZero.isSelected()) {
    // same logic with new InventoryRowModel(true)
}
if (radioWithMovement.isSelected()) {
    // same logic with new InventoryRowModel(true)
}
```

Moreover, I don't think we still need to manage the `specificRadio` button (because now is alone since the other one is just a button), the `codeTextField` (always active) should be enough with its `KeyListener` (see issue's screenshot example).

Reply:
oops sory I used now the right constructor for differents cases

Reply:
My comment included:

"Moreover, I don't think we still need to manage the `specificRadio` button because now it's the only JRadioButton since the other `selectButton` is now a standard JButton; the `codeTextField` (always active) should be enough with its `KeyListener` to allow the user to add at any time specific products to the inventory, so the JTextField + a JLabel is enough, see issue's screenshot as example."

Reply:
Moreover, this method has still a lot of duplicated code, are you sure it cannot be further refactored?